validator/client/src/main/java/tech/pegasys/teku/validator/client/doppelganger/DoppelgangerDetector.java

```diff
@@ -255,6 +285,22 @@ private SafeFuture<Void> checkValidatorsLivenessAtEpoch(
           .thenApply(doppelgangerDetected -> null);
     }
 
+    private void logMissingIndices(
+        Set<BLSPublicKey> pubKeys, Map<BLSPublicKey, UInt64> validatorIndicesByPubKey) {
```

Comment:
Finals please. Totally over-complication for this method, but let it avoid catching a sight

Reply:
Done